core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -127,6 +130,78 @@ public void register(BuiltinFunctionRepository repository) {
     repository.register(year());
   }
 
+  /**
+   * NOW() returns a constant time that indicates the time at which the statement began to execute.
+   * `fsp` argument support is removed until refactoring to avoid bug where `now()`, `now(x)` and
+   * `now(y) return different values.
+   */
+  private FunctionResolver now(FunctionName functionName) {
```

Comment:
Any reason to return `FunctionResolver`, but not `DefaultFunctionResolver`? Does this matter?

Reply:
`FunctionResolver` is an interface and `DefaultFunctionResolver` is an implementation. 
Standard practice is to return the most generic type possible.

Reply:
Should we groom then to return `FunctionResolver` everywhere in `DateTimeFunction`?